lib/srv/db/proxyserver.go

```diff
@@ -289,8 +297,8 @@ func (s *ProxyServer) MySQLProxy() *mysql.Proxy {
 // decoded from the client certificate by auth.Middleware.
 //
 // Implements common.Service.
-func (s *ProxyServer) Connect(ctx context.Context, user, database string) (net.Conn, *auth.Context, error) {
-	proxyContext, err := s.authorize(ctx, user, database)
+func (s *ProxyServer) Connect(ctx context.Context, params common.ConnectParams) (net.Conn, *auth.Context, error) {
```

Comment:
As we discussed before, we should respect limits on both proxy and database agent sides to account for various trusted cluster scenarios. So ProxyServer should also have a limiter. This Connect method would probably be a good place for it.

Reply:
Sorry, my mistake. Added.